Validator/UniqueValidator.php

```diff
@@ -0,0 +1,51 @@
+<?php
+namespace Vanio\DomainBundle\Validator;
+
+use Doctrine\ORM\EntityManager;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+class UniqueValidator extends ConstraintValidator
+{
+    /** @var EntityManager */
+    private $entityManager;
+
+    public function __construct(EntityManager $entityManager)
+    {
+        $this->entityManager = $entityManager;
+    }
+
+    public function validate($entity, Constraint $constraint)
+    {
+        if (!$constraint instanceof Unique) {
+            throw new UnexpectedTypeException($constraint, __NAMESPACE__.'\Unique');
+        }
+
+        $fields = $constraint->fields;
+        if (!is_array($fields)) {
+            $fields = [ $fields ];
+        }
+
+        $criteria = [];
+        foreach ($fields as $field) {
+            $criteria[$field] = $entity->$field();
+        }
+
+        $repository = $this->entityManager->getRepository($constraint->class);
+        $result = $repository->findBy($criteria);
+
+        if (count($result) == 0) {
+            return;
+        }
+
+        $id = $constraint->id;
+        if ($id && count($result) == 1 && $repository->get($entity->$id()) === current($result)) {
```

Comment:
Ta metoda get je jen v nasi repository , sice by tam mela byt pokud si to clovek spravne v configu nastavi aby to pouzivalo implementaci nasi repository, ale v tomhle pripade bude asi lepsi find, jedinej rozdil je v tom, ze find nevyhazuje vyjimku kdyz nic nenajde, ale vraci null, coz v tomhle pripade je naopak lepsi